src/theory/arith/constraint.cpp

```diff
@@ -701,109 +699,9 @@ bool Constraint::wellFormedFarkasProof() const {
-  RationalVector::const_iterator coeffBegin = cr.d_farkasCoefficients->begin();
-
-  while(antecedent != NullConstraint){
-    Assert(lhs.isNull() || Polynomial::isMember(lhs));
-
-    const Rational& coeff = *coeffIterator;
-    int coeffSgn = coeff.sgn();
-
-    rhs += antecedent->getValue() * coeff;
-
-    ArithVar antVar = antecedent->getVariable();
-    if(!lhs.isNull() && vars.hasNode(antVar)){
-      Node antAsNode = vars.asNode(antVar);
-      if(Polynomial::isMember(antAsNode)){
-        Polynomial lhsPoly = Polynomial::parsePolynomial(lhs);
-        Polynomial antPoly = Polynomial::parsePolynomial(antAsNode);
-        Polynomial sum = lhsPoly + (antPoly * coeff);
-        lhs = sum.getNode();
-      }else{
-        lhs = Node::null();
-      }
-    } else {
-      lhs = Node::null();
-    }
-    Debug("constraints::wffp") << "running sum: " << lhs << " <= " << rhs << endl;
-
-    switch( antecedent->getType() ){
-    case LowerBound:
-      // fc[l] < 0, therefore return false if coeffSgn >= 0
-      if(coeffSgn >= 0){ return false; }
-      break;
-    case UpperBound:
-      // fc[u] > 0, therefore return false if coeffSgn <= 0
-      if(coeffSgn <= 0){ return false; }
-      break;
-    case Equality:
-      if(coeffSgn == 0) { return false; }
-      break;
-    case Disequality:
-    default:
-      return false;
-    }
-    
-    if(coeffIterator == coeffBegin){ return false; }
-    --coeffIterator;
-    --p;
-    antecedent = d_database->d_antecedents[p];
-  }
-  if(coeffIterator != coeffBegin){ return false; }
-
-  const Rational& firstCoeff = (*coeffBegin);
-  int firstCoeffSgn = firstCoeff.sgn();
-  rhs += (getNegation()->getValue()) * firstCoeff;
-  if(!lhs.isNull() && vars.hasNode(getVariable())){
-    Node firstAsNode = vars.asNode(getVariable());
-    if(Polynomial::isMember(firstAsNode)){
-      Polynomial lhsPoly = Polynomial::parsePolynomial(lhs);
-      Polynomial firstPoly = Polynomial::parsePolynomial(firstAsNode);
-      Polynomial sum = lhsPoly + (firstPoly * firstCoeff);
-      lhs = sum.getNode();
-    }else{
-      lhs = Node::null();
-    }
-  }else{
-    lhs = Node::null();
-  }
-
-  switch( getNegation()->getType() ){
-  case LowerBound:
-    // fc[l] < 0, therefore return false if coeffSgn >= 0
-    if(firstCoeffSgn >= 0){ return false; }
-    break;
-  case UpperBound:
-    // fc[u] > 0, therefore return false if coeffSgn <= 0
-    if(firstCoeffSgn <= 0){ return false; }
-    break;
-  case Equality:
-    if(firstCoeffSgn == 0) { return false; }
-    break;
-  case Disequality:
-  default:
-    return false;
-  }
-  Debug("constraints::wffp") << "final sum: " << lhs << " <= " << rhs << endl;
-  // 0 = lhs <= rhs < 0
-  return (lhs.isNull() || (Constant::isMember(lhs) && Constant(lhs).isZero()))
-         && rhs.sgn() < 0;
-
-#else  /* IS_PROOFS_BUILD */
-  return true;
+  return cr.d_farkasCoefficients == RationalVectorCPSentinel;
 #endif /* IS_PROOFS_BUILD */
+  return true;
```

Comment:
Ditto: assume that `IS_PROOFS_BUILD` is true here.

Reply:
In the code block executed when `IS_PROOFS_BUILD` holds, the statement 
```
  return cr.d_farkasCoefficients == RationalVectorCPSentinel;
```
would be executed only if proofs are off, which we should assume is never the case, if I understand correctly. In that case I assume that the code that should survive is the one after, in the old code, `Assert(PROOF_ON());`?